crates/runtime-table-partition/src/provider.rs

```diff
@@ -109,8 +110,25 @@ impl PartitionTableProvider {
             partition_by,
             partitions,
             schema,
+            insert_strategy: Arc::new(DefaultInsertStrategy),
         })
     }
+
+    /// Creates a new [`PartitionTableProvider`] with a custom data insertion strategy.
+    ///
+    /// # Errors
+    /// This function will return an Error when the `partition_by` expression
+    /// validation fails.
+    pub async fn new_with_insert_strategy(
```

Comment:
Why not just add a `with_insert_strategy` instead of a new constructor?

Reply:
Updated, thank you